dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1002 @@
+              'resourceName', '')
+
+        # In case the message is an insert message
+        if log_message_type.endswith('insert') or log_message_type.endswith(
+            'createSnapshot'):
+          if request_metadata:
+            resource.creator_ip_address = request_metadata.get('callerIp')
+            resource.creator_useragent = request_metadata.get(
+                'callerSuppliedUserAgent')
+          self._ParseInsertLogMessage(resource, request, response)
+
+        elif log_message_type.endswith('delete'):
+          resource.deletion_timestamp = response.get('insertTime')
+          resource.deleted_by = response.get('user')
+          if request_metadata:
+            resource.deleter_ip_address = request_metadata.get('callerIp')
+            resource.deleter_useragent = request_metadata.get(
+                'callerSuppliedUserAgent')
+
+      else:
+        self.PublishMessage(f'Type {log_message_type} not supported')
+        resource = None
+
+      if resource:
+        if resource.id:
+          self.resources_dict[resource.id] = resource
+
+  def _ParseInsertLogMessage(self, resource: Resource, request: Dict[str, Any],
+                             response: Dict[str, Any]) -> None:
+    """Parses a GCP log message where the operation is insert or create.
+
+    Args:
+      resource: Resource object to update with parsed information
+      request: Request portion of the log message
+      response: Response portion of the log message
+    """
+    resource.creation_timestamp = response.get('insertTime')
+    resource.created_by = response.get('user', '')
+
+    if 'sourceDisk' in request:
+      if not resource.parent:
+        resource.parent = Resource()
+        resource.parent.resource_name = request.get('sourceDisk', '')
+
+    # When creating machine image from an instance
+    elif 'sourceInstance' in request:
+      if not resource.parent:
+        resource.parent = Resource()
+        resource.parent.resource_name = request.get('sourceInstance', '')
+
+    # When creating a disk image from another disk
+    elif 'sourceSnapshot' in request:
+      if not resource.parent:
+        resource.parent = Resource()
+        resource.parent.resource_name = request.get('sourceSnapshot', '')
+
+    # When creating a new instance, one of
+    # initializeParams.sourceImage or initializeParams.sourceSnapshot or
+    # disks.source is required except for local SSD.
+    if request.get('disks'):
+      for disk in request.get('disks', {}):
+        disk_resource = Resource()
+        disk_resource.project_id = resource.project_id
+        disk_resource.created_by = resource.created_by
+        disk_resource.creator_ip_address = resource.creator_ip_address
+        disk_resource.creator_useragent = resource.creator_useragent
+        disk_resource.zone = resource.zone
+        disk_resource.name = disk.get('deviceName')
+        disk_resource.type = 'gce_disk'
```

Comment:
consider sending values via the constructor while initializing the method, as it is easier to read
```suggestion
        disk_resource = Resource(
            name=disk.get('deviceName'),
            type='gce_disk',
            project_id=resource.project_id,
            zone=resource.zone,
            created_by=resource.created_by,
            creator_ip_address=resource.creator_ip_address,
            creator_useragent=resource.creator_useragent
        )
```

Reply:
I use the same style across, cause most of the time I keep adding to the Resource object as I parse data. So yes i can change that, but don't see the value and keeping it as is will be consistent across the code.

Reply:
@sa3eed3ed please check if I should mark this as resolved. Or do you want me to still do the change.